controllers/argocd/applicationset.go

```diff
@@ -224,9 +241,18 @@ func (r *ReconcileArgoCD) reconcileApplicationSetServiceAccount(cr *argoprojv1a1
 	}
 
 	if exists {
+		if applicationSetDisabled {
+			if err := r.Client.Delete(context.TODO(), sa); err != nil {
```

Comment:
can we have a `log.Info` here that mentions that the applicationSet service account is being deleted because applicationSet was disabled?